### ISSUE TITLE
Announce deployment on Gitter channels

### ISSUE THREAD

**Issue by ojongerius** [FreeCodeCamp/open-api#50]
We deploy automatically, to make this more transparent it would be great to announce deployment in 2 channels:

* For staging deployments, announce it in freecodecamp/open-api
* For production (master) deployments, announce in both freecodecamp/open-api and freecodecamp/contributors

**Comment by raisedadead** [comment]
We can configure our bot to do that. I am currently working on the github bot to use [probot](https://probot.github.io) and with that we can do all sorts of stuff.

**Comment by ojongerius** [comment]
@raisedadead sounds good to me! How would probot be notified of a deployment?

**Comment by raisedadead** [comment]
It can listen to events 'deploy hooks' and based on that it delivers a payload to the chat for announcement.

**Comment by ojongerius** [comment]
Aha, so let me recap for me to see if I understand this, and to check that we are on the same page:

**TL;DR** If  we use deployment events in Github, Probot will be able to handle those.

The flow that Github uses, see https://developer.github.com/v3/guides/delivering-deployments -they use Heathen to handle deployments:
* A webhook is added on the repository, and GH will send a POST to the configured endpoint 
* The endpoint receives the POST, sees if it is a event that requires a deployment, ie: a PR is merged to a specific branch
* The endpoint starts a deployment and sends a POST to Github that the deployment is done
* Probot will see the deployment event and can announce 

The flow for us, if we stick with Travis would be:
* When PRs are merged to deployable branches, a Travis build will run our deploy script is run and creates a deployment using the GH API 
 (see [create-a-deployment](https://developer.github.com/v3/repos/deployments/#create-a-deployment)).
* Probot announces a deployment has been created on the appropriate channels
* When the deployment is finished, depending on the outcome the script sets the deployment status using the GH API 
 (see [create-a-deployment-status](https://developer.github.com/v3/repos/deployments/#create-a-deployment-status)).
* Probot announces in what state the deployment finished

AFAICS work that needs doing is:

- [ ] Add GH API calls to create and update status of deployments to deploy script
- [ ] Configure actions for deployment events in Probot

@raisedadead how ready is Probot? How much work would it be to set this up? Something I can help with? 

/cc @Bouncey fyi, as this is related to https://github.com/freeCodeCamp/open-api/pull/44

**Comment by raisedadead** [comment]
I am keeping the Probot on back burner for now. I have started some work (just initial config) and stuff over at my fork of the [bot](https://github.com/freecodecamp/camper-gh-bot), I'll let you know as soon as its ready for PR.

Thanks for the summary, this will help us in future withe the progress.

**Comment by ojongerius** [comment]
There is a serverless plugin that allows pushing notification to Slack, or use webhooks. We can use Webhooks to announce to Gitter. https://github.com/maasglobal/serverless-plugin-notification

I'll see if I can get this working tomorrow morning.

**Comment by ojongerius** [comment]
Deployment are announced on Gitter. If we want to get a better experience than the "custom" webhook we use now, we'll have to either add that functionality to Gitter, use a bot or switch chat platform.

I might add some more data about the deployments in the current implementation, that would require a small change to https://github.com/maasglobal/serverless-plugin-notification
